### DISCUSSION TITLE
`serde` seems to have added a breaking change in `1.0.157`, would it have been detected by `cargo-semver-checks`?

### DISCUSSION THREAD

**Discussion by ginger51011** [obi1kenobi/cargo-semver-checks#423] [Q&A] [answered]
_See [this issue for `serde`](https://github.com/serde-rs/serde/issues/2415)_

So I've been casually pulling out my hair trying to figure out where my problem was (first blaming another poor crate, or `rustc`), and then it turns out it was `serde` releasing what ended up being (or looks to me) to be a breaking change in a patch version. Doing `=1.0.156` fixes it, but still.

Would `serde` have been saved by `cargo-semver-checks`, or do we need a new query?

**Comment by obi1kenobi** [comment, marked as answer]
I don't think _currently_ `cargo-semver-checks` would have helped here, but it's probable that a new lint could catch this.

I'll keep an eye on that issue. Once we see what the fix looks like (or the code that broke it), we'll know for sure what the lint rule should look for.

Thanks for bringing this to my attention, I probably would have missed it otherwise!

**Reply by ginger51011** [reply]
It was closed now in 1.0.159, see [this PR](https://github.com/serde-rs/serde/pull/2422)

**Reply by obi1kenobi** [reply]
Unfortunately, it seems like it's a non-trivial change. I was hoping it might be something macro-rules-like which we might have been able to parse and check. But instead it boils down to finding a behavior change in the output of a bunch of complex Rust code, which isn't as easy. I don't think it can be done in a cost-effective way unfortunately.

**Reply by ginger51011** [reply]
As I suspected. Oh well, thanks for taking a look!

**Comment by obi1kenobi** [comment]
Thanks for mentioning it!
